### ISSUE TITLE
preventOverlapping not working on closures

### ISSUE THREAD

**Issue by g19i85g** [lavary/crunz#314]
**Crunz version**: 2.2.0

**PHP version**: 7.3.19

**Description**  
The method preventOverlapping() does not work.

**How to reproduce**  
Create a long running task and let it run every minute.
After a few minutes you can see that the task runs several times.
```php
use Crunz\Schedule;

$schedule = new Schedule();
$task = $schedule->run(
    static function () {
        sleep(500);

        return true;
    }
);
$task->everyMinute()
     ->preventOverlapping();

return $schedule;
```

**Additional context**  
I found out that the method \Crunz\Event::lockKey() creates a new key every time I start the run command.

```php
public function serialize()
    {
        ...
        $this->reference = spl_object_hash($this->closure);
       ...
        $ret = \serialize(array(
            ...
             'self' => $this->reference,
         ));
       ...
    }
```
I suppose the part of the method \Opis\Closure\SerializableClosure::serialize() is the cause.

**Comment by PabloKowalczyk** [comment]
Hello, after https://github.com/lavary/crunz/issues/313 you can specify `lockKey` yourself, that should resolve the problem.

**Comment by g19i85g** [comment]
Yes, that's true but from my point of view this is more of a workaround. Maybe you could make it mandatory to hand over the key when using closures.

**Comment by PabloKowalczyk** [comment]
Fixed by #326, thanks @g19i85g.
